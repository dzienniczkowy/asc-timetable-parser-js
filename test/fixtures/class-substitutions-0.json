{
  "r": "<div class=\"_RSS357\"><style type=\"text/css\">._RSS357 .signature {font-Size:20px;float:right;border-Bottom:1px dashed #c0c0c0;width:100px;margin-Right:4px;margin-Left:10px}\n._RSS357 .row {display:flex;flex-Direction:row;align-Items:baseline;font-Size:12px;margin-Bottom:1px}\n._RSS357 .period {font-Size:16px;text-Align:center;flex-Grow:0;margin-Right:10px}\n._RSS357 .period>span {white-Space:pre;display:inline-block;min-Width:30px}\n._RSS357 .info {flex:1}\n._RSS357 .section {margin-Bottom:15px;border:1px solid #c0c0c0;border-Radius:6px;padding:4px}\n._RSS357 .header {font-Size:20px;margin-Bottom:5px;font-Weight:bold;border:2px solid #c0c0c0;border-Radius:6px;padding:4px;display:inline-block}\n._RSS357 .rows {padding-Left:5px}\n._RSS357 .rows .signature {font-Size:inherit}</style><div data-date=\"2022-09-08\"><div style=\"text-align:center\"><span class=\"print-font-resizable\">Absent teachers: Dąbrowski Wawrzyniec  (0 - 4), Adamski Bolek , Szczepańska Wiesława (3) </span></div><div style=\"height:15px;flex:none\"></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1A</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">3</span></div><div class=\"info\"><span class=\"print-font-resizable\">(historia i teraźniejszość) ➔ język polski - Substitution: (Dąbrowski Wawrzyniec) ➔ Mazurek Bogumiła</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1D</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(0)</span></div><div class=\"info\"><span class=\"print-font-resizable\">informatyka_1: informatyka - Adamski Bolek, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1E</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">5</span></div><div class=\"info\"><span class=\"print-font-resizable\">język polski - Substitution: (Szczepańska Wiesława) ➔ Kopeć Mateusz</span></div></div><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">6</span></div><div class=\"info\"><span class=\"print-font-resizable\">(język polski) ➔ historia - Substitution: (Szczepańska Wiesława) ➔ Piotrowska Marta</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">1F</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">3</span></div><div class=\"info\"><span class=\"print-font-resizable\">(język polski) ➔ edukacja dla bezpieczeństwa - Substitution: (Szczepańska Wiesława) ➔ Lewandowski Maksymilian</span></div></div><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">4</span></div><div class=\"info\"><span class=\"print-font-resizable\">(język polski) ➔ matematyka - Substitution: (Szczepańska Wiesława) ➔ Grzelak Ewelina</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">2F</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1)</span></div><div class=\"info\"><span class=\"print-font-resizable\">język polski - Szczepańska Wiesława, Cancelled</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">3A</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">4</span></div><div class=\"info\"><span class=\"print-font-resizable\">(wiedza o społeczeństwie) ➔ podstawy przedsiębiorczości - Substitution: (Dąbrowski Wawrzyniec) ➔ Witkowska Alicja</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">4D</span></div><div class=\"rows\"><div class=\"row change\"><div class=\"period\"><span class=\"print-font-resizable\">3</span></div><div class=\"info\"><span class=\"print-font-resizable\">grupa 2: informatyka - Substitution: (Adamski Bolek) ➔ Piasecka Dorota, zł. grup</span></div></div></div></div><div class=\"section print-nobreak\"><div class=\"header\"><span class=\"print-font-resizable\">4G</span></div><div class=\"rows\"><div class=\"row remove\"><div class=\"period\"><span class=\"print-font-resizable\">(1 - 2)</span></div><div class=\"info\"><span class=\"print-font-resizable\">informatyka_2: informatyka - Adamski Bolek, Cancelled</span></div></div></div></div></div></div>"
}
