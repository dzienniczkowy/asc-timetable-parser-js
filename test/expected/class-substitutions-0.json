{
  "date": "2022-09-08",
  "absentClasses": [],
  "absentTeachers": [
    {
      "name": "Dąbrowski Wawrzyniec",
      "lessons": {
        "first": 0,
        "last": 4
      }
    },
    {
      "name": "Adamski Bolek",
      "lessons": null
    },
    {
      "name": "Szczepańska Wiesława",
      "lessons": {
        "first": 3,
        "last": 3
      }
    }
  ],
  "sections": [
    {
      "name": "1A",
      "changes": [
        {
          "lessons": {
            "first": 3,
            "last": 3
          },
          "info": "(historia i teraźniejszość) ➔ język polski - Substitution: (Dąbrowski Wawrzyniec) ➔ Mazurek Bogumiła",
          "cancelled": false
        }
      ]
    },
    {
      "name": "1D",
      "changes": [
        {
          "lessons": {
            "first": 0,
            "last": 0
          },
          "info": "informatyka_1: informatyka - Adamski Bolek, Cancelled",
          "cancelled": true
        }
      ]
    },
    {
      "name": "1E",
      "changes": [
        {
          "lessons": {
            "first": 5,
            "last": 5
          },
          "info": "język polski - Substitution: (Szczepańska Wiesława) ➔ Kopeć Mateusz",
          "cancelled": false
        },
        {
          "lessons": {
            "first": 6,
            "last": 6
          },
          "info": "(język polski) ➔ historia - Substitution: (Szczepańska Wiesława) ➔ Piotrowska Marta",
          "cancelled": false
        }
      ]
    },
    {
      "name": "1F",
      "changes": [
        {
          "lessons": {
            "first": 3,
            "last": 3
          },
          "info": "(język polski) ➔ edukacja dla bezpieczeństwa - Substitution: (Szczepańska Wiesława) ➔ Lewandowski Maksymilian",
          "cancelled": false
        },
        {
          "lessons": {
            "first": 4,
            "last": 4
          },
          "info": "(język polski) ➔ matematyka - Substitution: (Szczepańska Wiesława) ➔ Grzelak Ewelina",
          "cancelled": false
        }
      ]
    },
    {
      "name": "2F",
      "changes": [
        {
          "lessons": {
            "first": 1,
            "last": 1
          },
          "info": "język polski - Szczepańska Wiesława, Cancelled",
          "cancelled": true
        }
      ]
    },
    {
      "name": "3A",
      "changes": [
        {
          "lessons": {
            "first": 4,
            "last": 4
          },
          "info": "(wiedza o społeczeństwie) ➔ podstawy przedsiębiorczości - Substitution: (Dąbrowski Wawrzyniec) ➔ Witkowska Alicja",
          "cancelled": false
        }
      ]
    },
    {
      "name": "4D",
      "changes": [
        {
          "lessons": {
            "first": 3,
            "last": 3
          },
          "info": "grupa 2: informatyka - Substitution: (Adamski Bolek) ➔ Piasecka Dorota, zł. grup",
          "cancelled": false
        }
      ]
    },
    {
      "name": "4G",
      "changes": [
        {
          "lessons": {
            "first": 1,
            "last": 2
          },
          "info": "informatyka_2: informatyka - Adamski Bolek, Cancelled",
          "cancelled": true
        }
      ]
    }
  ]
}
